internal/pkg/agent/cmd/container.go

```diff
@@ -771,9 +771,10 @@ func logToStderr(cfg *configuration.Configuration) {
 
 func setPaths(statePath, configPath, logsPath, socketPath string, writePaths bool) error {
 	statePath = envWithDefault(statePath, "STATE_PATH")
-	if statePath == "" {
+	if isStatePathTooLong(statePath) || statePath == "" {
```

Comment:
I'm not sure this is correct, people change the state path to mount persistent volumes to the container so that the filebeat registry and fleet state aren't ephemeral. Silently doing something else isn't right.

It appears all you need to do imove where the control socket gets created, why didn't you do that instead?

We already have a helper for that, does it not work in the container file system? 

https://github.com/elastic/elastic-agent/blob/414ae849ebd5d67b9f6e369753607083a6c013f9/pkg/utils/socket_unix.go#L21-L30

Reply:
> I'm not sure this is correct, people change the state path to mount persistent volumes to the container so that the filebeat registry and fleet state aren't ephemeral. Silently doing something else isn't right.

That did cross my mind when creating this PR. The Agent failing to start didn't seem a good behaviour either.

> It appears all you need to do imove where the control socket gets created, why didn't you do that instead?

It didn't work, the Agent was filing to find the configuration file, the config path and state path can be the same if `CONFIG_PATH` is not set: https://github.com/elastic/elastic-agent/blob/d79b4ef6887959833e8fea0d31dc5c1a5678d1d1/internal/pkg/agent/cmd/container.go#L778-L781

I don't know much about this part of the Elastic-Agent, so I tried to keep the changes to a minimum.

I can go back dig deeper on why I got failures when changing only the socket path.

Reply:
We need to do something else other than silently not use the state path, this will lead to extremely surprising behavior for some people on Kubernetes. Failing to start the agent is probably more correct than doing this as it tells the user they have something to fix.

Reply:
I managed to get the container command to work with the socket path outside of the state path. Now we explicitly create the config path (before it was only the top path).

I hope nothing outside this function makes any assumption about the socket.

Reply:
If you exec into the container and run the `elastic-agent status` command does it work?